sigv4-signing/go/Sigv4Signer.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"strings"
+	"time"
+)
+
+const (
+	ALGORITHM_AWS4_HMAC_SHA_256 = "AWS4-HMAC-SHA256"
+	AWS4_REQUEST_TYPE           = "aws4_request"
+	SERVICE                     = "kinesisvideo"
+	X_AMZ_ALGORITHM             = "X-Amz-Algorithm"
+	X_AMZ_CREDENTIAL            = "X-Amz-Credential"
+	X_AMZ_DATE                  = "X-Amz-Date"
+	X_AMZ_EXPIRES               = "X-Amz-Expires"
+	X_AMZ_SECURITY_TOKEN        = "X-Amz-Security-Token"
+	X_AMZ_SIGNATURE             = "X-Amz-Signature"
+	X_AMZ_SIGNED_HEADERS        = "X-Amz-SignedHeaders"
+	NEW_LINE_DELIMITER          = "\n"
+	DATE_PATTERN                = "20060102"
+	TIME_PATTERN                = "20060102T150405Z"
+	METHOD                      = "GET"
+	SIGNED_HEADERS              = "host"
+)
+
+type AwsV4Signer struct{}
+
+// SigV4 Signer sample reference for the Kinesis Video Streams WebRTC WebSocket connections
+// Uses Gorilla WebSockets Go Module
+
+func main() {
+
+	////////////////////////////////////////////
+	// ***Sample connection test:*** - fill in the variables:
+
+	// Connect as Viewer URL - GetSignalingChannelEndpoint + Query Parameters Channel ARN as X-Amz-ChannelARN + Client Id as X-Amz-ClientId
```

Comment:
Ref: https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_SingleMasterChannelEndpointConfiguration.html#KinesisVideo-Type-SingleMasterChannelEndpointConfiguration-Role

Reply:
Can we add a note that the Signlaing Endpoint responses wlil be unique for each  ConnectAsMaster and ConnectAsViewer operations?

Reply:
Added